### ISSUE TITLE
[Leo]: Lift hardcoded client-size context limits

### ISSUE THREAD

**Issue by LorenzoMinto** [brave/brave-browser#41620]
### Description

Lift client-size context size limits. We are moving the limit enforcement on the server side. We will still keep some upper limits to cap the amount of MB sent over. 

Limits to lift:
- input box 
- page content

**Comment by petemill** [comment]
@LorenzoMinto In practice the only thing we're really limiting client-side is the page content size. Does this mean we should no longer do the text embedder "get top similarity" for the prompt? cc @darkdh

**Comment by LorenzoMinto** [comment]
Yes, full context and local RAG address different use cases cc @darkdh @bbondy to confirm. I think that feature is still behind a flag and disabled in prod, we were looking for a way to surface it in a way that would restrain it to QA but haven't converged on one. 

We can kill two birds with one stone and address the input box context limit too with this PR.

**Comment by evrenesat** [comment]
Hi, is that means when I use Leo with a local LLM with a big enough context size, I will no longer get `The page is too long for Leo. Some context could be missing from the conversation. Leo was able to read 15% of the page's contents.` warning ?

**Comment by mattmcalister** [comment]
Let's bump the input box limit up to 10k characters (at minimum) which would support a 2k word report. 10k characters (incl spaces) = ~2-3k tokens.

**Comment by LaurenWags** [comment]
Requires 1.81.108 or higher for verification 👍🏻

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.81.110 Chromium: 138.0.7204.97 (Official Build) beta (64-bit)
-- | --
Revision | 7ac186a03520c9f07fcad0941c161dfa899a5ab3
OS | Windows 11 Version 24H2 (Build 26100.4349)
```

Verified using the STR from the issue description https://github.com/brave/brave-browser/issues/41620#issue-2586546254

#### Confirmed Leo input length is now set to 20k character limit instead of 2000
#### Confirmed once input length is above 20k input can't be submitted


1.80.x | 1.81.x 
----- | -----
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/fb9925c7-d126-4f73-abfa-eefea93d5c0b" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/6226a0ac-40d0-4ed1-afe2-a8c0cd78e897" />


https://github.com/user-attachments/assets/9b6b93ed-bffb-46fb-9c88-6eb76ba61b26

**Comment by LaurenWags** [comment]
Requires 1.80.123 or higher for verification 👍🏻 

Removed `QA Pass-Win64` label as this will need to be re-verified on 1.80.x.

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.80.123 Chromium: 138.0.7204.157 (Official Build) (64-bit)
-- | --
Revision | 4724a9bf79e2fa4c8405789625f37d865484bb4f
OS | Windows 11 Version 24H2 (Build 26100.4349)
JavaScript | V8 13.8.258.26
```

Verified using the STR from the issue description https://github.com/brave/brave-browser/issues/41620#issue-2586546254

#### Confirmed Leo input length is now set to 20k character limit instead of 2000
#### Confirmed once input length is above 20k input can't be submitted
#### Successfully submitted when the input length is under 20k

https://github.com/user-attachments/assets/000b7aac-9da4-46e2-9cd3-71b635bba050
